easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -320,6 +325,48 @@ def configure_step(self, srcdir=None, builddir=None):
 
         return out
 
+    def _get_cmake_python_config(self):
+        """Get the CMake configuration options for Python hints."""
+        options = {}
+        python_root = get_software_root('Python')
+        if python_root:
+            python_version = LooseVersion(get_software_version('Python'))
+            python_exe = os.path.join(python_root, 'bin', 'python')
+            # This is required for (deprecated) `find_package(PythonInterp ...)`
+            options['PYTHON_EXECUTABLE'] = python_exe
+            # Ensure that both `find_package(Python) and find_package(Python2/3)` work as intended
+            options['Python_EXECUTABLE'] = python_exe
+            if python_version >= "3":
+                options['Python3_EXECUTABLE'] = python_exe
+            else:
+                options['Python2_EXECUTABLE'] = python_exe
+        return options
+
+    def get_cmake_python_config_dict(self):
+        """Get a dictionary with the CMake configuration options for Python hints."""
+        return self._get_cmake_python_config()
```

Comment:
I don't see a need for this function if it does nothing but forwarding, so they are the same function.

Reply:
Implemented with [fa30b9c](https://github.com/easybuilders/easybuild-easyblocks/pull/3463/commits/fa30b9cd6c40ec1a02db44b802c6cb39525ec338)

Reply:
But there are still 2 functions where `get_cmake_python_config_dict  == _get_cmake_python_config`. I'd just use one.

Reply:
It was to have them more modularized in case in the future there is need for more separation between the dict and str functions, but i guess for now they can just be one